server/uploader.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * @fileoverview upload function handler
+ */
+
+'use strict'
+
+var path;
+var fs = require('fs');
+var multer = require('multer');
+var shell = require('shelljs');
+
+var utils = require('./utils');
+var binding = require('./binding');
+var segtree = require('./segtree');
+
+var upload = multer();
+
+module.exports = {
+
+  /**
+   * Upload a file or a directory to server.
+   * @param req {Object} Including query parameders.
```

Comment:
variable name is after the type spec, i.e. "@param {Object} req Including query parameters."
{string} and {number} are lower-case.
{Array} and {Object} are upper-case.